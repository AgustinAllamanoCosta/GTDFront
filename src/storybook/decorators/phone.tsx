import { GoogleOAuthProvider } from '@react-oauth/google';
import { styled } from 'styled-components';
import { Children } from 'react';

export const PhoneContext = (Story: any) => (
  <AppContext>
    <Story />
  </AppContext>
);

type AppContextProps = {
  children: string | JSX.Element | JSX.Element[];
};

export const AppContext = ({ children }: AppContextProps) => {
  const key: string = process.env.VITE_CLEINT_ID
    ? process.env.VITE_CLEINT_ID
    : '';
  return (
    <GoogleOAuthProvider clientId={key}>
      <MyPhoneContext>{children}</MyPhoneContext>
    </GoogleOAuthProvider>
  );
};

const MyPhoneContext = styled.div`
  background-color: black;
  display: flex;
  align-items: center;
  justify-content: center;
  width: 390px;
  height: 884px;
`;
