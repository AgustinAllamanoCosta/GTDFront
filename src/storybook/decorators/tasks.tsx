import { v4 as uuidv4 } from 'uuid';
import { Task } from '../../types/types';
import { BrowserRouter } from 'react-router-dom';
import ItemsContext from '../../components/useItems/useItems';

const task1: Task = {
  title: 'some task 1',
  isComplete: false,
  id: uuidv4(),
  isCancele: false,
  isActive: true,
};
const task2: Task = {
  title: 'some task 2',
  isComplete: false,
  id: uuidv4(),
  isCancele: false,
  isActive: true,
};
const task3: Task = {
  title: 'some task 3',
  isComplete: false,
  id: uuidv4(),
  isCancele: false,
  isActive: true,
};

export const mockTaskInContext: Array<Task> = [task1, task2, task3];

export const TaskContextDecorator = (Story: any) => {
  return (
    <BrowserRouter>
      <ItemsContext>
        <Story />
      </ItemsContext>
    </BrowserRouter>
  );
};
