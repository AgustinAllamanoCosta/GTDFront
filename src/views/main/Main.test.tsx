import React from "react";
import { expect } from '@jest/globals';
import { render, act, screen } from "@testing-library/react";
import { AppContext} from "../../storybook/decorators/phone";
import * as credentials from "../../assets/google/client_secret_153375467669-bl3575q27tmq97cf359v0unoegk06opn.apps.googleusercontent.com.json";
import MainView from "./Main";

describe("Login View", () => {

  test("Should render a login view with singin and login buttons", async () => {

    const { container } = render(
      <AppContext  credential={credentials}>
        <MainView/>
      </AppContext>
    );
    expect(container).toMatchSnapshot();
  });
});
