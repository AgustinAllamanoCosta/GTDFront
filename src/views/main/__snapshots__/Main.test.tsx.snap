// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login View Should render a login view with singin and login buttons 1`] = `
<div>
  <div
    class="sc-hfDuwB wnNtd"
  >
    <div
      class="sc-hbwYeq kaJZKU"
    >
      <div
        class="sc-dBwcjI bhYBxT"
      >
        <div
          class="sc-kPNBzt kfRNst"
          data-cy="Avatar-photo"
          imageurl="https://i.stack.imgur.com/Dj7eP.jpg"
        />
        <div
          class="sc-jCIOXC fNsQyj"
        >
          <div
            class="sc-fBEEOr fKCOYR"
            data-cy="Card-Contents"
          >
            <div
              class="sc-bXxmBy bSIEXq"
              data-cy="User-Card-Header"
            >
              <div
                class="sc-dMnqAs cAOBYD"
              >
                <div
                  class="sc-jlOLRj jEvdST"
                >
                  <span
                    class="sc-laEPQd hnPnfI"
                    data-cy="Card-title"
                  >
                    Getting Things Done
                  </span>
                </div>
              </div>
              <span
                class="sc-eErDDp WdbYy"
                data-cy="Card-SubTitle"
              >
                Hi Agustin Allamano Costa !
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-jCIOXC fNsQyj"
      >
        <div
          class="sc-fBEEOr eFWqUg"
          data-cy="Card-Contents"
        >
          <div
            class="sc-evVYyG dsYGDa"
          >
            <span
              class="sc-BOZFl hhMcHb"
              data-cy="Card-title"
            >
              Active Task
            </span>
            <span
              class="sc-BOZFl hhMcHb"
              data-cy="Card-label"
            >
              0/0
            </span>
          </div>
          <div
            class="sc-cpKBeH brLBgc"
          />
        </div>
      </div>
      <div
        class="sc-jCIOXC fNsQyj"
      >
        <div
          class="sc-fBEEOr eFWqUg"
          data-cy="Card-Contents"
        >
          <div
            class="sc-evVYyG dsYGDa"
          >
            <span
              class="sc-BOZFl hhMcHb"
              data-cy="Card-title"
            >
              Task Inbox
            </span>
            <span
              class="sc-BOZFl hhMcHb"
              data-cy="Card-label"
            >
              total 0
            </span>
          </div>
          <div
            class="sc-jjIKjs sc-daVtvr bMktBm kyGhnj"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-plus sc-kbUMEY kFwCnV"
              data-icon="plus"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 80c0-17.7-14.3-32-32-32s-32 14.3-32 32V224H48c-17.7 0-32 14.3-32 32s14.3 32 32 32H192V432c0 17.7 14.3 32 32 32s32-14.3 32-32V288H400c17.7 0 32-14.3 32-32s-14.3-32-32-32H256V80z"
                fill="currentColor"
              />
            </svg>
            <input
              class="sc-bubcaR gMPofY"
              placeholder="Add Task"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
